### PR TITLE
FenicsUfcx 0.4.1

### PR DESCRIPTION
**Pull request by ma595**
Renamed Ufcx to FenicsUfcx and added FenicsUfcx dependency to FenicsDolfinx to remove python3 requirement.

### CODE REVIEW

**Comment by ma595** [comment]
Build still fails due to [PR #484](https://github.com/FEniCS/ffcx/pull/484.

File: var/spack/repos/builtin/packages/fenics-dolfinx/package.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ class FenicsDolfinx(CMakePackage):
 
     depends_on("fenics-ufcx", type=("build"))
     depends_on("fenics-ufcx@main", type=("build"), when="@main")
-    depends_on("fenics-ufcx@0.4.0", type=("build"), when="@0.4.0")
+    depends_on("fenics-ufcx@0.4.1", type=("build"), when="@0.4.1")
```
Comment (garth-wells) [inline var/spack/repos/builtin/packages/fenics-dolfinx/package.py:46]: Maybe this should be
```python
depends_on("fenics-ufcx@0.4.1:", type=("build"), when="@0.4.1")
```
(note the `:`) ?

I think the type should be `type=("build", "link")`. This is the Spack default [1], so `type` could even be dropped.

[1] https://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-types

File: var/spack/repos/builtin/packages/fenics-ufcx/package.py [hunk 2]
```diff
@@ -6,17 +6,18 @@
...
     """
 
     homepage = 'https://github.com/FEniCS/ffcx'
     git = 'https://github.com/FEniCS/ffcx.git'
-    url = 'https://github.com/FEniCS/ffcx/archive/refs/heads/main.zip'
+    url = 'https://github.com/FEniCS/ffcx/archive/v0.4.0.tar.gz'
     maintainers = ['ma595']
 
     version('main', branch='main')
+    version('0.4.0', sha256='2145286576304cf303f2c94dbea3bd7de0d2140797202d9504c0d581eea9cbc1')
```
Comment (garth-wells) [inline var/spack/repos/builtin/packages/fenics-ufcx/package.py:21]: Need to make this 0.4.1.

**Comment by garth-wells** [comment]
Should we change ParMETIS to required, and make Scotch a variant?

**Comment by ma595** [comment]
> Should we change ParMETIS to required, and make Scotch a variant?

I've made these changes to fenics-dolfinx. Is this what you had in mind?

**Comment by chrisrichardson** [comment]
Does this branch work for `spack install basix@0.3.0` ? I've a feeling it may fail, due to changes in build process...

**Comment by ma595** [comment]
> Does this branch work for `spack install basix@0.3.0` ? I've a feeling it may fail, due to changes in build process...

spack install py-fenics-basix@0.3.0 seems to work for me.

**Review by chrisrichardson — APPROVED** [review]

**Pull request closed without merging**